util/promlint/promlint.go

```diff
@@ -205,6 +210,54 @@ func lintHistogramSummaryReserved(mf dto.MetricFamily) []Problem {
 	return problems
 }
 
+// lintMetricTypeInName detects when the type of the metric is included in the metric name.
+func lintMetricTypeInName(mf dto.MetricFamily) []Problem {
+	t := mf.GetType()
+	if t == dto.MetricType_UNTYPED {
+		return nil
+	}
+
+	var problems problems
+	n := strings.ToLower(mf.GetName())
+
+	typename := strings.ToLower(t.String())
+	if strings.Contains(n, "_"+typename+"_") || strings.HasSuffix(n, "_"+typename) {
+		problems.Add(mf, fmt.Sprintf(`%s metrics should not include the type in metric name`, typename))
+	}
+	return problems
+}
+
+// lintReservedChars detects colons in metric names.
+func lintReservedChars(mf dto.MetricFamily) []Problem {
+	var problems problems
+	if strings.Contains(mf.GetName(), ":") {
+		problems.Add(mf, "metric names should not contain ':'")
+	}
+	return problems
+}
+
+// lintCamelCase detects metric names written in camelCase.
+func lintCamelCase(mf dto.MetricFamily) []Problem {
+	var problems problems
+	re := regexp.MustCompile(`[a-z][A-Z]`)
```

Comment:
As a nit: I'd pull this out to a package level var like 'var camelCaseRe = regexp.MustCompile(`[a-z][A-Z]`)' and then use it in this function.

Reply:
I want to keep the locality - definition near the use, but happy to pull out to package var